fasm2bels/models/clk_models.py

```diff
@@ -108,6 +108,44 @@ def process_bufg(conn, top, tile, features):
         top.add_site(site)
 
 
+def cleanup_hrow(top, site):
+    """
+    Cleans-up BUFHCE if one is configured as pass-throu
+    """
+
+    # Check if we have a BUFHCE
+    bel = site.maybe_get_bel("BUFHCE")
+    if bel is not None:
+
+        # Get source for CE
+        source = top.find_source_from_sink(site, 'CE')
+        # Get CE inversion
+        inv_ce = bel.parameters['IS_CE_INVERTED']
+
+        # Determine if the BUFHCE is permanently enabled
+        if source in [0, 1] and (source ^ inv_ce) == 1:
+
+            # Prune site routing
+            site.prune_site_routing(('site_pin', 'I'))
+            site.prune_site_routing(('site_pin', 'CE'))
+            site.prune_site_routing(('bel_pin', 'BUFHCE', 'O', 'output'))
+
+            # Remove the BUFHCE source. This is needed as the bel cannot be
+            # removed when connected to the source (clearing bel.connections
+            # doesn't solve that).
+            del site.sources['O']
+
+            # Remove the BUFHCE bel connection and the bel itself
+            bel.connections = {}
+            top.remove_bel(site, bel)
+
+            # Link site input and output
+            site.link_site_routing([
+                ('site_pin', 'I'),
+                ('site_pin', 'O'),
+            ])
```

Comment:
Because you are do the link after site intergration, you need to add the short, see here: https://github.com/SymbiFlow/symbiflow-xc-fasm2bels/blob/47c46270de32418a5f7a69e0db3e4d54db9849bc/fasm2bels/models/verilog_modeling.py#L1520-L1532

Maybe refactor that logic to be invokable during clean up?

Reply:
I can try. Does shorting nets requires a PIP? Because after BUFHCE removal there is no single PIP that could connect those nets.

Reply:
> I can try. Does shorting nets requires a PIP? Because after BUFHCE removal there is no single PIP that could connect those nets.

That's not true?  `CLK_HROW_BOT_R/CLK_HROW_BOT_R.CLK_HROW_CK_MUX_OUT_L10->>CLK_HROW_CK_HCLK_OUT_L10` ?

Reply:
Ok.